From eaca65e0e0d982e175f3224720588d0b4575f965 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Thu, 1 Apr 2021 00:06:54 +0200
Subject: [PATCH 3/4] Fix docdir for Gentoo

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index bb65543..1896d9a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -99,7 +99,7 @@ mandir = @mandir@
 man1ext = 1
 man1dir = $(mandir)/man$(man1ext)
 datadir = @datadir@
-docdir = @docdir@@VERSIONED_PATH@
+docdir = @docdir@
 
 # The following must be substituted using @datadir@ and @libdir@
 # to avoid adding RPM generation paths (CUPS STR #1112)
-- 
2.33.1

