diff -up file-5.42/src/file.c.omv~ file-5.42/src/file.c
--- file-5.42/src/file.c.omv~	2022-06-14 14:41:32.708884703 +0200
+++ file-5.42/src/file.c	2022-06-14 14:40:40.436535597 +0200
@@ -509,7 +509,7 @@ unwrap(struct magic_set *ms, const char
 
 	if (strcmp("-", fn) == 0) {
 		f = stdin;
-		wid = 1;
+		wid = 0;
 	} else {
 		if ((f = fopen(fn, "r")) == NULL) {
 			file_warn("Cannot open `%s'", fn);
@@ -530,7 +530,7 @@ unwrap(struct magic_set *ms, const char
 	while ((len = getline(&line, &llen, f)) > 0) {
 		if (line[len - 1] == '\n')
 			line[len - 1] = '\0';
-		e |= process(ms, line, wid);
+		e |= process(ms, line, (wid > 0) ? wid : file_mbswidth(ms, line));
 	}
 
 	free(line);
