diff -up Python-3.7.2/Tools/gdb/libpython.py.p179~ Python-3.7.2/Tools/gdb/libpython.py
--- Python-3.7.2/Tools/gdb/libpython.py.p179~	2018-12-23 22:37:36.000000000 +0100
+++ Python-3.7.2/Tools/gdb/libpython.py	2018-12-26 00:02:34.238488684 +0100
@@ -307,6 +307,9 @@ class PyObjectPtr(object):
                 self.tp_name = tp_name
                 self.address = address
 
+            def __len__(self):
+                return len(repr(self))
+
             def __repr__(self):
                 # For the NULL pointer, we have no way of knowing a type, so
                 # special-case it as per
@@ -959,6 +962,8 @@ class PyFrameObjectPtr(PyObjectPtr):
         try:
             with open(os_fsencode(filename), 'r') as fp:
                 lines = fp.readlines()
+        except TypeError: # filename is FakeRepr
+            return None
         except IOError:
             return None
 
@@ -1667,9 +1672,12 @@ class Frame(object):
 
     def print_summary(self):
         if self.is_evalframe():
-            pyop = self.get_pyop()
-            if pyop:
+            try:
+                pyop = self.get_pyop()
                 line = pyop.get_truncated_repr(MAX_OUTPUT_LEN)
+            except:
+                pyop = None
+            if pyop:
                 write_unicode(sys.stdout, '#%i %s\n' % (self.get_index(), line))
                 if not pyop.is_optimized_out():
                     line = pyop.current_line()
@@ -1686,7 +1694,10 @@ class Frame(object):
 
     def print_traceback(self):
         if self.is_evalframe():
-            pyop = self.get_pyop()
+            try:
+                pyop = self.get_pyop()
+            except:
+                pyop = None
             if pyop:
                 pyop.print_traceback()
                 if not pyop.is_optimized_out():
