--- Python-3.6.2/Modules/posixmodule.c.omv~	2017-07-17 22:05:54.713633142 +0200
+++ Python-3.6.2/Modules/posixmodule.c	2017-07-17 23:45:52.205621039 +0200
@@ -8614,7 +8614,7 @@ static unsigned int
 os_major_impl(PyObject *module, dev_t device)
 /*[clinic end generated code: output=5b3b2589bafb498e input=1e16a4d30c4d4462]*/
 {
-    return major(device);
+    return gnu_dev_major(device);
 }
 
 
@@ -8631,7 +8631,7 @@ static unsigned int
 os_minor_impl(PyObject *module, dev_t device)
 /*[clinic end generated code: output=5e1a25e630b0157d input=0842c6d23f24c65e]*/
 {
-    return minor(device);
+    return gnu_dev_minor(device);
 }
 
 
