diff -up Python-3.8.0/setup.py.p4~ Python-3.8.0/setup.py
--- Python-3.8.0/setup.py.p4~	2019-11-03 02:45:16.301940776 +0100
+++ Python-3.8.0/setup.py	2019-11-03 02:45:43.046884257 +0100
@@ -814,7 +814,7 @@ class PyBuildExt(build_ext):
             self.missing.append('spwd')
 
         # select(2); not on ancient System V
-        self.add(Extension('select', ['selectmodule.c']))
+        self.add(Extension('select', ['selectmodule.c'], libraries=['m']))
 
         # Fred Drake's interface to the Python parser
         self.add(Extension('parser', ['parsermodule.c']))
