--- file-5.31/magic/Localstuff.0009~	2017-05-25 13:12:53.369806955 +0200
+++ file-5.31/magic/Localstuff	2017-05-25 13:13:45.350495525 +0200
@@ -5,3 +5,9 @@
 # $File: Localstuff,v 1.5 2007/01/12 17:38:27 christos Exp $
 # Add any locally observed files here.  Remember:
 # text if readable, executable if runnable binary, data if unreadable.
+
+# File magic for Xen, the virtual machine monitor for x86
+0      string          LinuxGuestRecord        Xen saved domain
+#>2    regex           \(name\ [^)]*\)         %s
+>20    search/256      (name                   (name
+>>&1   string          x                       %s...)
