From 73bce48536158ba7a3ccb06aa86e2e85ce2b9881 Mon Sep 17 00:00:00 2001
From: brvphoenix <vanwaholtz@gmail.com>
Date: Fri, 27 May 2022 22:02:29 +0800
Subject: [PATCH] Don't decompress the reply data for qt 6.3

---
 src/base/net/downloadhandlerimpl.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/base/net/downloadhandlerimpl.cpp b/src/base/net/downloadhandlerimpl.cpp
index b2de6213ace..476836d6119 100644
--- a/src/base/net/downloadhandlerimpl.cpp
+++ b/src/base/net/downloadhandlerimpl.cpp
@@ -121,9 +121,13 @@ void DownloadHandlerImpl::processFinishedDownload()
     }
 
     // Success
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
+    m_result.data = m_reply->readAll();
+#else
     m_result.data = (m_reply->rawHeader("Content-Encoding") == "gzip")
                     ? Utils::Gzip::decompress(m_reply->readAll())
                     : m_reply->readAll();
+#endif
 
     if (m_downloadRequest.saveToFile())
     {
